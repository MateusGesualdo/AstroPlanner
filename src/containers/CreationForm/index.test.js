import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import { CreationForm, Form, Input } from './index'

describe("Testes do formulário", () => {
    
    test("Snapshot", () => {

        const tree = renderer
            .create(<CreationForm/>)
            .toJSON()

        expect(tree).toMatchSnapshot()

    })

    test("Deve chamar a funcao handleSubmission ao clicar no botão", () => {
        
        const handleSubmission = jest.fn()
        const component = shallow(<Form  onSubmit={handleSubmission}/> )
                
        component.simulate("submit")

		expect(handleSubmission).toHaveBeenCalledTimes(1)
    })  

    test("Deve chamar a funcao setTask ao digitar uma nova tarefa", () => {
        
        const setTask = jest.fn()
        const component = shallow(<Form ><Input onChange={setTask}/></Form> )
        const input = component.find(Input)
               
        input.simulate("change")

		expect(setTask).toHaveBeenCalledTimes(1)
    })  

})