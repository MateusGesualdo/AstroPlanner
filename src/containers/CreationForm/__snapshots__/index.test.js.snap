// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do formulário Snapshot 1`] = `
<form
  className="sc-bwzfXH bRcjrU classes.root"
  onSubmit={[Function]}
>
  <input
    className="sc-htpNat dJGdEm"
    name="text"
    onChange={[Function]}
    placeholder="Nova Tarefa"
    required={true}
  />
  <select
    name="day"
    onChange={[Function]}
  >
    <option
      value=""
    >
      Selecione um dia
    </option>
    <option>
      Segunda
    </option>
    <option>
      Terça
    </option>
    <option>
      Quarta
    </option>
    <option>
      Quinta
    </option>
    <option>
      Sexta
    </option>
    <option>
      Sábado
    </option>
    <option>
      Domingo
    </option>
  </select>
  <button
    className="sc-bdVaJa edZuZW"
    type="submit"
  >
    Salvar
  </button>
</form>
`;
